### PR TITLE
my solution

### PR DESCRIPTION
**Pull request by viktalfrontend**
[DEMO LINK](https://viktalfrontend.github.io/js_2048_game/)

### CODE REVIEW

File: src/modules/Game.class.js [hunk 1]
```diff
@@ -20,25 +21,204 @@ class Game {
...
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    this.score = 0;
+    this.status = 'idle';
     // eslint-disable-next-line no-console
     console.log(initialState);
```
Comment (TarasHoliuk) [inline src/modules/Game.class.js:96]: remove it

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,7 +1,109 @@
...
+    game.restart();
+    resetButtonUpdate();
+    updateField();
+  }
+});
+
+document.addEventListener('keydown', (e) => {
+  e.preventDefault();
+
+  if (game.status === GameStatus.WIN || game.status === GameStatus.LOSE) {
```
Comment (TarasHoliuk) [inline src/scripts/main.js:87]: I can start playing even if didn't click start button yet:

image

File: src/modules/Game.class.js [hunk 3]
```diff
@@ -20,25 +21,204 @@ class Game {
...
+      filteredRow = filteredRow.filter(num => num !== 0);
+
+      while (filteredRow.length < 4) {
+        filteredRow.push(0);
+      }
+
+      this.state[row] = filteredRow;
+    }
+
+    this.addRandomNumber();
```
Comment (TarasHoliuk) [inline src/modules/Game.class.js:121]: you shouldn't add a new number if the field didn't change after button click

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]
Have you tried to fix tests? If no at least try it. Ask for help in the chat if you need 🙂

**Comment by viktalfrontend** [comment]
until no one helped to correct the tests, I am sending the work as it is

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
There are a lot of tests failing. Try to pass them. Feel free to ask for help in the chat

**Review by VitaliyBondarenko1982 — APPROVED** [review]
Good job.
Works correctly.

**Pull request closed without merging**